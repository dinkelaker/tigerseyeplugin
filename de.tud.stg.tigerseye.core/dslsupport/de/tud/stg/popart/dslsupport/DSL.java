package de.tud.stg.popart.dslsupport;

/**
 * Marker interface. Identifies a DSL defining class.
 *
 */
public interface DSL {

}
