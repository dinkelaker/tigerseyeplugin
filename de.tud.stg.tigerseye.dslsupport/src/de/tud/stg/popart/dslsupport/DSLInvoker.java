/**
 * Copyright 2008, Darmstadt University of Technology
 * GNU GENERAL PUBLIC LICENSE version 2.0
 * @author Tom Dinkelaker
 **/
package de.tud.stg.popart.dslsupport;

import groovy.lang.Closure;
import groovy.lang.MissingMethodException;
import groovy.lang.MissingPropertyException;

import java.util.List;
import java.util.Set;

import org.codehaus.groovy.runtime.InvokerHelper;

public class DSLInvoker {

	private final static boolean DEBUG = false; 
//	================================
//	Methods below are no longer used
//	================================
//	/**
//	 * Tries to invoke the first found keyword method in DSL definition list.
//	 */
//	public static Object invokeMethodOnDslDefs(List dslDefinitions, String name, Object args) {
//		if (DEBUG) println "DSLInvoker.invokeMethodOnDslDefs(): \t try to invoke "+name+" "+args+" on dslDefs";
//		Object result = null;
//		boolean found = false;
//		
//		Object dslDefinition = null;
//
//		Iterator it = dslDefinitions.iterator();
//        while (it.hasNext()) {
//			dslDefinition = it.next();
//			if (DEBUG) println "DSLInvoker.invokeMethodOnDslDefs(): \t try to invoke "+name+" "+args+" on "+dslDefinition;
//		    try {
//		    	//result = tryInvokeMethodOnDslDefinition(dslDefinition, name, args);
//		    	result = InvokerHelper.invokeMethod(dslDefinition,name, args);
//		    	found = true;
//		    } catch (MissingMethodException e1) {
//		    	//do nothing try next dsl
//		    }
//	    	if (found) break;
//		}
//		if (!found) {
//		    //if (DEBUG) println "DSLInvoker.invokeMethodOnDslDefs(): \t not found "+name+" "+args+" in dslDefs.";
//			//throw new MissingMethodException(name,DSLInvoker.class,args);
//			throw new RuntimeException(); //TODO Check exception
//		} else {
//  	        if (DEBUG) println "DSLInvoker.invokeMethodOnDslDefs(): \t found "+name+" "+args+" in "+dslDefinition.class;
//		    return result;
//		}
//	}
//
//	/**
//	 * Tries to invoke the keyword method on a DSL definition.
//	 */
//	private static Object tryInvokeMethodOnDslDefinition(Object dslDefinition, String name, Object args) {
//	    if (DEBUG) println "DSLInvoker.tryInvokeMethodOnDslDefinition(): \t try to invoke on "+name+" "+dslDefinition.class;
//	    try {
//	    	Object result = InvokerHelper.invokeMethod(dslDefinition,name, args);
//		    if (DEBUG) println "DSLInvoker.tryInvokeMethodOnDslDefinition(): \t found "+name+"="+result+" in "+dslDefinition.class;
//	        return result
//		} catch (MissingMethodException e1) {
//		    if (DEBUG) println "DSLInvoker.tryInvokeMethodOnDslDefinition(): \t not found "+name+" "+args+" on "+dslDefinition.class;
//			//do nothing an try next dsl definition
//		}
//	}
//	
//	/**
//	 * Tries to get the first found keyword literal in DSL definition list.
//	 */
//	public static Object getPropertyOnDslDefs(List dslDefinitions, String _property) {
//		Object result = null;
//		boolean found = false;
//		
//		Object dslDefinition = null;
//
//		Iterator it = dslDefinitions.iterator();
//		if (DEBUG && (!it.hasNext())) println "DSLInvoker: empty DSL list";
//        while (it.hasNext()) {
//			dslDefinition = it.next();
//			if (DEBUG) println "DSLInvoker.getPropertyOnDslDefs(): \t try to invoke "+_property+" on "+dslDefinition.class;
//		    try {
//		    	//result = tryGetPropertyOnDslDefinition(dslDefinition, _property);
//		    	result = InvokerHelper.getProperty(dslDefinition,_property);
//		    	found = true;
//		    } catch (MissingPropertyException e1) {
//				//do nothing an try next dsl definition
//		    }
//	    	if (found) break;
//		}
//		if (!found) {
//		    //if (DEBUG) println "DSLInvoker.invokeMethodOnDslDefs(): \t not found "+_property+" in dslDefs.";
//			//throw new MissingPropertyException(_property,DSLInvoker.class);
//			throw new RuntimeException();
//		} else {
//  	        if (DEBUG) println "DSLInvoker.getPropertyOnDslDefs(): \t found "+_property+" in "+dslDefinition.class;
//		    return result;
//		}
//	}
//		
//	/**
//	 * Tries to get the keyword literal in DSL definition.
//	 */
//	private static Object tryGetPropertyOnDslDefinition(Object dslDefinition, String _property){
//	    if (DEBUG) println "DSLInvoker.tryGetPropertyOnDslDefinition(): \t try getting "+_property+" from "+dslDefinition.class;
//		
//		try {
//			Object result = InvokerHelper.getProperty(dslDefinition,_property);
//		    if (DEBUG) println "DSLInvoker.tryGetPropertyOnDslDefinition(): \t found "+_property+"="+result+" in "+dslDefinition.class;
//		    return result;
//		} catch (MissingPropertyException e00) {
//			//do nothing an try next dsl definition
//		}
//	}
//	================================
//	Methods above are no longer used
//	================================
	
	/**
	 * Zips the result of all keyword methods in DSL definition list.
	 */
	public static Object zipMethodOnDslDefs(Set<DSL> dslDefinitions, Closure zipWithClosure, String name, Object args) {
		if (DEBUG) System.out.println("DSLInvoker.zipMethodOnDslDefs(): \t zips method results of "+name+" "+args+" on dslDefs");
		assert zipWithClosure.getMaximumNumberOfParameters() == dslDefinitions.size();

		List results = new java.util.LinkedList(); 
		

		for(Object dslDefinition : dslDefinitions){
			if (DEBUG) System.out.println("DSLInvoker.zipMethodOnDslDefs(): \t try to invoke "+name+" "+args+" on "+dslDefinition);
		    try {
		    	Object result = InvokerHelper.invokeMethod(dslDefinition,name, args);
		    	results.add(result);
		    } catch (MissingMethodException e1) {
		    	results.add(null);
		    }
		}
		
		if (DEBUG) System.out.println("DSLInvoker.zipMethodOnDslDefs(): \t results=$results");
        Object result = zipWithClosure.call(results.toArray());	
        if (DEBUG) System.out.println("DSLInvoker.zipMethodOnDslDefs(): \t final result=$result");
        return result;
	}
	
	/**
	 * Zips the value of all keyword literal in DSL definition list.
	 */
	public static Object zipPropertyOnDslDefs(Set<DSL> dslDefinitions, Closure zipWithClosure, String _property) {
		if (DEBUG) System.out.println("DSLInvoker.zipMethodOnDslDefs(): \t zips property values of "+_property+" on dslDefs");
		assert zipWithClosure.getMaximumNumberOfParameters() == dslDefinitions.size();
		
		List results = new java.util.LinkedList();
		
		for(Object dslDefinition : dslDefinitions){
			if (DEBUG) System.out.println("DSLInvoker.zipPropertyOnDslDefs(): \t try to invoke "+_property+" on "+dslDefinition.getClass());
		    try {
		    	Object result = InvokerHelper.getProperty(dslDefinition,_property);
		    	results.add(result);
		    } catch (MissingPropertyException e1) {
				results.add(null);
		    }
		}
  	    if (DEBUG) System.out.println("DSLInvoker.zipPropertyOnDslDefs(): \t results="+results);
  	    Object result = zipWithClosure.call(results.toArray());
  	  	if (DEBUG) System.out.println("DSLInvoker.zipPropertyOnDslDefs(): \t final result="+result);
		return result;
	}
		
}
