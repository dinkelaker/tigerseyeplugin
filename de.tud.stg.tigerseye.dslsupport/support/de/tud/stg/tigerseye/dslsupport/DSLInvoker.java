package de.tud.stg.tigerseye.dslsupport;

import groovy.lang.Closure;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashMap;
import java.util.HashSet;
import java.util.List;
import java.util.Set;
import java.util.concurrent.ConcurrentHashMap;

import de.tud.stg.tigerseye.dslsupport.logger.DSLSupportLogger;

public final class DSLInvoker{
	
	private static final String NO_CONTEXT = "no_context_information";

	private static final DSLSupportLogger logger = new DSLSupportLogger(DSLInvoker.class);
	
	// TODO For Issue #37 (Leo_Roos;Nov 18, 2011)
	// In the current version, the DSL instances are shared between different DSL files.
	// Shouldn't cache instances, they might change between files, need additional information about the file _and_ class to have an exact instance definition
	private static final ConcurrentHashMap<dslObjectKey<?>, DSL> invokers = new ConcurrentHashMap<dslObjectKey<?>, DSL>();

	public static Object eval(Class<? extends DSL> clazz, Closure<?> c) {
		return eval(NO_CONTEXT,toList(clazz),c);
	}
	
	public static Object eval(String contextInformation, Class<? extends DSL> clazz, Closure<?> c) {
		return eval(contextInformation, toList(clazz), c);
	}

	public static Object eval(Class<? extends DSL>[] clazzes, Closure<?> c) {
		return eval(NO_CONTEXT,Arrays.asList(clazzes) , c);		
	}
	
	public static Object eval(List<Class<? extends DSL>> clazzes, Closure<?> c) {
		return eval(NO_CONTEXT, clazzes, c);
	}
	
	public static Object eval(String contextInformation, List<Class<? extends DSL>> clazzes, Closure<?> c) {
		
		Set<DSL> dsls = new HashSet<DSL>();

		for (Class<? extends DSL> clazz : clazzes) {
			DSL dsl = getDSL(clazz);
			if (dsl != null)
				dsls.add(dsl);
		}

		InterpreterCombiner ic = new InterpreterCombiner(dsls, new HashMap<String, Object>()); //TODO For Issue #36, sort the DSL instances "dsls"
		return ic.eval(c);
	}
	
	public static <T extends DSL> T getDSL(Class<T> clazz) {
		return getDSL(NO_CONTEXT, clazz);
	}

	public static <T extends DSL> T getDSL(String contextInformation, Class<T> clazz) {
		return getDSL(new dslObjectKey<T>(contextInformation, clazz));
	}

	private static List<Class<? extends DSL>> toList(Class<?>... clazzes) {
		List<Class<? extends DSL>> result = new ArrayList<Class<? extends DSL>>(clazzes.length);
		for (Class<?> c : clazzes) {
			/* Throws exception if not of type DSL */
			Class<? extends DSL> clazz = (Class<? extends DSL>) c;
			result.add(clazz);
		}
		return result;
	}

	private static <T extends DSL> T getDSL(dslObjectKey<T> invokerKey) {				
		DSL object = invokers.get(invokerKey);

		if (object == null) {
			try {
				object = invokerKey.clazz.newInstance();
			} catch (InstantiationException e) {
				logger.error("Failed to instantiate class", e);
			} catch (IllegalAccessException e) {
				logger.error("Failed to instantiate class", e);
			}
			
			invokers.put(invokerKey, object);
		}
		
		if(object == null){
			throw new IllegalArgumentException("No instance for "+ invokerKey + " was found, nor could it be instantiated.");
		}
		
		return (T) object;
	}

	static DSL getInvoker(Class<? extends DSL> clazz) {
		return getDSL(clazz);
	}
	
	private static class dslObjectKey<T>{
		
		private final String context;
		private final Class<T> clazz;

		public dslObjectKey(String context, Class<T> clazz) {
			this.context = context;
			this.clazz = clazz;
		}
		
		@Override
		public boolean equals(Object obj) {
		    	if (obj == null){
		    		return false;
		    	}
		    	if (obj == this){
		    		return true;
		    	}
		    	if (obj.getClass() != this.getClass()) {
		    		return false;
		    	}
		    	DSLInvoker.dslObjectKey<?> other = (DSLInvoker.dslObjectKey<?>) obj;
		    	boolean equal = context == other.context && clazz.equals(other.clazz); 
		    	return equal;
		    }
		
		@Override
		public int hashCode() {
			return context.hashCode() * 11 + 7  * clazz.hashCode();
		}
		
		@Override
		public String toString() {
			return "DSLObjectKey[context="+context+",class="+clazz+"]";
		}
	}
	
/*
	private Class<? extends DSL> clazz;

	private DSLInvoker() {
		//Not supposed to be instantiated
	}

	/**
	 * @deprecated use static methods for consistent access
	 //
	@Deprecated
	public DSLInvoker(Class<? extends DSL> clazz) {
		this.clazz = clazz;
	}

	/**
	 * @deprecated use static methods for consistent access
	 
	@Deprecated
	public Object eval(Closure<?> c) {
		if (this.clazz == null)
			throw new IllegalStateException("If instance eval is used the Invoker must have been initialized with a clazz");
		else
			return eval(this.clazz, c);
	}
	*/
	
}
