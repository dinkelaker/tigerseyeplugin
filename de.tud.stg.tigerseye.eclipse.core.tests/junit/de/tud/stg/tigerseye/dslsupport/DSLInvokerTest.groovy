package de.tud.stg.tigerseye.dslsupport;

import static org.junit.Assert.*;
import static de.tud.stg.tigerseye.eclipse.core.utils.CustomFESTAssertions.*;

import org.junit.Before;
import org.junit.Ignore;
import org.junit.Test;

import de.tud.stg.popart.builder.test.dsls.MathDSL;
import de.tud.stg.popart.builder.test.dsls.SimpleSqlDSL;

public class DSLInvokerTest {


	
	@Test
	public void shouldThrowExceptionIfOnlyClosure() throws Exception {
		def res = DSLInvoker.eval(MathDSL.class, {
			def p = sum__p0(
					[10, 10]as int[])
		})
		//Result is last from closure
		assertEquals(20,res)
	}

	@Test
	public void shouldTransformSimpleDSLStatement() throws Exception {
		def res = DSLInvoker.eval(MathDSL.class, {
			sum__p0(
					[10, 10]as int[])
		})
		//Result is last from closure
		assertEquals(20,res)
	}

	@Test
	public void shouldTransformCombination() throws Exception {
		int spy = 0;
		def res = DSLInvoker.eval([MathDSL.class, SimpleSqlDSL.class],{
			spy = sum__p0(
					[10, 10]as int[])
			selectFrom(
					[
						"id",
						"age",
						"haircolor"
					]
					as String[],
					["students"
					]as String[])
		})
		//Last statement was selectFrom which returns the generated Query
		assertThat(res).contains("id");
		assertThat(res).contains("age");
		assertThat(res).contains("haircolor");
		assertThat(res).contains("students");
		assertEquals(20,spy) 
	}
	
	@Test
	public void shouldEvalClosureWithoutSpecialSyntax() throws Exception {
		int i = DSLInvoker.eval(MathDSL.class) {   def i = 1 + 1  }
		assertEquals(2, i)
	}
	
	@Ignore("Does not tell anything about Tigerseye")
	@Test
	public void shouldFailIfRunAsGroovyScript() throws Exception {
		//Expected test to fail because that File run as a Groovy Script via context menu will cause a MissingMethodException
		GroovyScriptEngine gse = new GroovyScriptEngine(".");
		File script = new File("junit/"+ RunAsGroovyScriptFails.class.getName().replaceAll("\\.", "/") + ".groovy")
		String form = script.toString()
		String res = gse.run(form, null)
		assertThat(res).contains("why you no run as groovy script?");
	}
}
