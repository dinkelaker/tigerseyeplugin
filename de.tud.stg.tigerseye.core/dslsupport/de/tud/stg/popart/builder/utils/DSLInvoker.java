package de.tud.stg.popart.builder.utils;

import groovy.lang.Closure;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashMap;
import java.util.LinkedList;
import java.util.List;
import java.util.Map;

import de.tud.stg.popart.dslsupport.DSL;
import de.tud.stg.popart.dslsupport.InterpreterCombiner;

public class DSLInvoker {
	private static final DSLInvoker instance = new DSLInvoker();

	private static final Map<Class<? extends DSL>, DSL> invokers = new HashMap<Class<? extends DSL>, DSL>();

	private final Class<? extends DSL> clazz;

	public static <T extends DSL> T getDSL(Class<T> clazz) {

		DSL object = invokers.get(clazz);

		if (object == null) {
			try {
				object = clazz.newInstance();
			} catch (InstantiationException e) {
				e.printStackTrace();
			} catch (IllegalAccessException e) {
				e.printStackTrace();
			}
			invokers.put(clazz, object);
		}

		return (T) object;

	}

	private DSLInvoker() {
		this.clazz = null;
	}

	public DSLInvoker(Class<? extends DSL> clazz) {
		this.clazz = clazz;
	}

	public static DSLInvoker getDSLInvoker() {
		return instance;
	}

	public static DSL getInvoker(Class<? extends DSL> clazz) {
		return getDSL(clazz);
	}

	public Object eval(Closure c) {
		if (this.clazz != null)
			return eval(this.clazz, c);
		else
			throw new IllegalStateException();
	}

	public static Object eval(Class<? extends DSL> clazz, Closure c) {
		DSL dsl = getDSL(clazz);

		c.setDelegate(dsl);
		c.setResolveStrategy(Closure.DELEGATE_FIRST);
		return c.call();
	}

	public static Object eval(Class<? extends DSL>[] clazzes, Closure c) {
		List<DSL> dsls = new LinkedList<DSL>();

		for (Class<? extends DSL> clazz : clazzes) {
			dsls.add(getDSL(clazz));
		}

		InterpreterCombiner ic = new InterpreterCombiner(dsls, new HashMap());
		return ic.eval(c);
	}

	public static <T> T[] asArray(T... ts) {
		return ts;
	}

	public static <T> List<T> asList(T... ts) {
		ArrayList<T> list = new ArrayList<T>(ts.length);
		for (T t : ts) {
			list.add(t);
		}
		return list;
	}

	public static void main(String[] args) {
		List<Integer>[] asArray = asArray(new ArrayList<Integer>(), new ArrayList<Integer>(), new ArrayList<Integer>());

		List<ArrayList<Integer>> asList = Arrays.asList(new ArrayList<Integer>(), new ArrayList<Integer>(),
				new ArrayList<Integer>());

	}
}
