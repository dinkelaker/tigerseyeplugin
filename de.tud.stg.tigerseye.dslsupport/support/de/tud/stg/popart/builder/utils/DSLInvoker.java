package de.tud.stg.popart.builder.utils;

import groovy.lang.Closure;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.LinkedList;
import java.util.List;
import java.util.Map;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import de.tud.stg.popart.dslsupport.DSL;
import de.tud.stg.popart.dslsupport.InterpreterCombiner;

public class DSLInvoker {
	
	private static final Logger logger = LoggerFactory
			.getLogger(DSLInvoker.class);
	
	private static final DSLInvoker instance = new DSLInvoker();

	private static final Map<Class<? extends DSL>, DSL> invokers = new HashMap<Class<? extends DSL>, DSL>();

	public static <T> T[] asArray(T... ts) {
		return ts;
	}

	public static <T> List<T> asList(T... ts) {
		ArrayList<T> list = new ArrayList<T>(ts.length);
		for (T t : ts) {
			list.add(t);
		}
		return list;
	}

	public static Object eval(Class<? extends DSL> clazz, Closure c) {
		DSL dsl = getDSL(clazz);

		c.setDelegate(dsl);
		c.setResolveStrategy(Closure.DELEGATE_FIRST);
		return c.call();
	}

	public static Object eval(Class<? extends DSL>[] clazzes, Closure c) {
		List<DSL> dsls = new LinkedList<DSL>();

		for (Class<? extends DSL> clazz : clazzes) {
			dsls.add(getDSL(clazz));
		}

	/*
	 * FIXME(Leo Roos): Must use the @deprecated method until the runtime
	 * jars, which also package a version of an InterpreterCombiner, are
	 * generated from the same source this class accesses during
	 * development.
	 */
	InterpreterCombiner ic = new InterpreterCombiner(dsls,
		new HashMap<String, Object>());
		return ic.eval(c);
	}

	public static <T extends DSL> T getDSL(Class<T> clazz) {

		DSL object = invokers.get(clazz);

		if (object == null) {
			try {
				object = clazz.newInstance();
			} catch (InstantiationException e) {
				logger.error("Failed to instantiate class", e);
			} catch (IllegalAccessException e) {
				logger.error("Failed to instantiate class", e);
			}
			invokers.put(clazz, object);
		}

		return (T) object;

	}

	public static DSLInvoker getDSLInvoker() {
		return instance;
	}

	public static DSL getInvoker(Class<? extends DSL> clazz) {
		return getDSL(clazz);
	}

	private final Class<? extends DSL> clazz;

	private DSLInvoker() {
		this.clazz = null;
	}

	public DSLInvoker(Class<? extends DSL> clazz) {
		this.clazz = clazz;
	}

	public Object eval(Closure c) {
		if (this.clazz != null)
			return eval(this.clazz, c);
		else
			throw new IllegalStateException("Closure must not be null");
	}

}
