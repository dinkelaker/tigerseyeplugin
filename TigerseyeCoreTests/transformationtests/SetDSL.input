package de.tud.stg.popart.builder.test.generated;

import de.tud.stg.popart.builder.utils.DSLInvoker;
import de.tud.stg.popart.builder.test.dsls.SetDSL;
	 
new DSLInvoker(SetDSL.class).eval() {
	String a = "a";
	
	Set M = {"a", "b", "c"};
	
	print M
}