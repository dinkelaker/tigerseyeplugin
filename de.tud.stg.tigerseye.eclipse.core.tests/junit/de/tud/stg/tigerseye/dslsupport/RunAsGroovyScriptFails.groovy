package de.tud.stg.tigerseye.dslsupport

import de.tud.stg.popart.builder.test.dsls.MathDSL;
import de.tud.stg.tigerseye.dslsupport.DSLInvoker



DSLInvoker.eval(MathDSL.class){
    return "why you no run as groovy script?"
    }