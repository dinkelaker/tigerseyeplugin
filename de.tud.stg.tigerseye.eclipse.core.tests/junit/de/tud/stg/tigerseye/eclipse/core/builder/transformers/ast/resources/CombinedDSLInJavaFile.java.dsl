package de.tud.stg.tigerseye.eclipse.core.builder.transformers.ast.resources;

import java.util.Set;

import de.tud.stg.popart.builder.test.dsls.SetDSL;
import de.tud.stg.popart.builder.test.dsls.SimpleSqlDSL;
import de.tud.stg.tigerseye.dslsupport.DSLInvoker;

public class CombinedDSLInJavaFile {
	
	public CombinedDSLInJavaFile() {
	
	}
	
	public void foo() { 
		
		Set s = {"k", "l"} â‹ƒ {"m", "n"};
		
		System.out.println("Should return \"[n, l, m, k]\" was: " + s);
		
		String query = SELECT "id" FROM "students";
		
		System.out.println("Query was: " + query);
	}
	
	public static void main(String[] args) {
		new CombinedDSLInJavaFile().foo();
	}
}