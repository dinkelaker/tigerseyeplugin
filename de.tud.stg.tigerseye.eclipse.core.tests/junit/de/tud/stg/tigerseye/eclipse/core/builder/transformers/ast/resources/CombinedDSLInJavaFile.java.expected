package de.tud.stg.tigerseye.eclipse.core.builder.transformers.ast.resources;

import java.util.Set;

import de.tud.stg.popart.builder.test.dsls.SetDSL;
import de.tud.stg.popart.builder.test.dsls.SimpleSqlDSL;
import de.tud.stg.tigerseye.dslsupport.DSLInvoker;

public class CombinedDSLInJavaFile {

	public CombinedDSLInJavaFile() {

	}

	public void foo() {

		Set s = DSLInvoker.getDSL(SetDSL.class).union(
				DSLInvoker.getDSL(SetDSL.class).asSet(
						DSLInvoker.getDSL(SetDSL.class).multiElementedList("k",
								DSLInvoker.getDSL(SetDSL.class).singleElementedList("l"))),
				DSLInvoker.getDSL(SetDSL.class).asSet(
						DSLInvoker.getDSL(SetDSL.class).multiElementedList("m",
								DSLInvoker.getDSL(SetDSL.class).singleElementedList("n"))));

		System.out.println("Should return \"[n, l, m, k]\" was: " + s);

		String query = DSLInvoker.getDSL(SimpleSqlDSL.class).selectFrom(new String[] { "id" },
				new String[] { "students" });

		System.out.println("Query was: " + query);

	}

	public static void main(String[] args) {
		new CombinedDSLInJavaFile().foo();
	}
}